src/api/observability-api.ts

```diff
@@ -372,6 +397,25 @@ export function withHeliconeOpenAI(
  * The ObservabilityAPI class provides methods to wrap API calls with Helicone logging
  */
 export class ObservabilityAPI extends BasePaymentsAPI {
+  protected readonly heliconeBaseLoggingUrl: string
+  protected readonly heliconeManualLoggingUrl: string
+
+  constructor(options: PaymentOptions) {
+    super(options)
+
+    // TODO: For testing purposes only. Remove once helicone is deployed to staging
+    // Get Helicone API key from environment variable and override the base class property
+    this.heliconeApiKey = process.env.HELICONE_API_KEY ?? this.heliconeApiKey
```

Comment:
@nova-florealis we cannot throw an error here if helicone key is not defined or else the payments library would not be able to initialize since this constructor is called.